core/domain/skill_domain.py

```diff
@@ -653,3 +653,48 @@ def to_dict(self):
             'skill_model_created_on': self.skill_model_created_on,
             'skill_model_last_updated': self.skill_model_last_updated
         }
+
+
+class SkillsMastery(object):
+    """Domain object for an Oppia Skills Mastery."""
+
+    @classmethod
+    def get_model_id(cls, user_id, skill_id):
+        """Returns model id corresponding to user and skill.
+
+        Args:
+            user_id: str. The user id of the logged in user.
+            skill_id: str. The unique id of the skill.
+
+        Returns:
+            str. The model id corresponding to user and skill.
+        """
+        return user_id + '-' + skill_id
```

Comment:
Generally we use interpolation: '%s-%s' % (user_id, skill_id).

That said, I think IDs can contain dashes. You'll probably want to use a colon as the delimiter instead.

Reply:
> That said, I think IDs can contain dashes. You'll probably want to use a colon as the delimiter instead.

Did you mean IDs can't contain dashes? This would suffice : '%s:%s' % (user_id, skill_id)?

Reply:
Yes, that's right. If you look at the id generation I believe the IDs come from the base64 charset, which includes dashes. So your composite ID would be ambiguous when it comes to parsing it into its components.